### DISCUSSION TITLE
Add support for images with floating points formats

### DISCUSSION THREAD

**Discussion by victor-hugo-dc** [microsoft/psi#172] [Ideas]
Currently, Psi does not offer support for images with floating point formats. Having support for these types of images could be very beneficial for the community as it would allow for depth images captured in ROS to be loaded into Psi directly. For example, this includes depth images captured by the Kinect ROS driver, which are output in 32FC1 format. We might be able to contribute this feature. Did we miss any existing solutions or there is any known issues that prevent its implementation in the first place?

**Comment by danbohus** [comment]
You are right. The framework does not yet have support for that image format, but it would be beneficial and we would welcome a contribution on this. If you decide you want to pursue this with the goal of creating a pull request, please create an associated issue and we can use it to discuss a bit more the design. For instance, it seems one could add a new `PixelFormat`, make sure the various imaging operators work correctly (or throw appropriate exceptions) over images of this type, and perhaps the `DepthImage` class could be modified to support not only the currently supported unsigned int format, but also the proposed float format. Would have to also consider updating the various classes involved in visualization.
